lib/spack/spack/binary_distribution.py

```diff
@@ -2296,7 +2295,7 @@ def download_buildcache_entry(file_descriptions, mirror_url=None):
         mirror_root = os.path.join(mirror_url, _build_cache_relative_path)
         return _download_buildcache_entry(mirror_root, file_descriptions)
 
-    for mirror in spack.mirror.MirrorCollection().values():
+    for mirror in spack.mirror.MirrorCollection(binary=True).values():
```

Comment:
(minor, comment) Given the amount of times we do:
```python
spack.mirror.MirrorCollection(binary=True).values()
```
I wonder if we should have a convenience wrapper:
```python
def binary_mirrors():
   return MirrorCollection(binary=True).values()
```
and use that in this module.

Reply:
We mix MirrorCollection(...) and MirrorCollection(...).values(), and these objects get passed on conditionally to other functions, etc, so I'm not very comfortable changing this right now.

Reply:
Ok, we can check this later and submit a PR if it makes sense.